pilot/pkg/security/authz/model/generator.go

```diff
@@ -180,33 +181,33 @@ func (srcNamespaceGenerator) principal(_, value string, _ bool, useAuthenticated
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-type srcServiceAccountGenerator struct{}
+type srcServiceAccountGenerator struct {
+	policyName types.NamespacedName
+}
 
-func (srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
+func (g srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
 	return nil, fmt.Errorf("unimplemented")
 }
 
-func (srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
-	regex, err := serviceAccountRegex(value)
-	if err != nil {
-		return nil, err
-	}
+func (g srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
+	regex := serviceAccountRegex(g.policyName.Namespace, value)
 	m := matcher.StringMatcherRegex(regex)
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-func serviceAccountRegex(value string) (string, error) {
+func serviceAccountRegex(defaultNamespace string, value string) string {
 	ns, sa, ok := strings.Cut(value, "/")
 	if !ok {
-		return "", fmt.Errorf("invalid value: %v", value)
+		ns = defaultNamespace
```

Comment:
If `!ok` do we want to enforce that `defaultNamespace` is non empty?

Reply:
Kind of but I am not super sure what we can do. Its a definitely a coding bug if its unset but we don't really have a `debug_assert!()` like in rust/C/etc...